overcooked_ai_py/agents/agent.py

```diff
@@ -515,3 +532,2155 @@ def ml_action(self, state):
+    - motion goals can be retained rather than re-finding goals every timestep
+    - make the wrong decision about a motion goal with prob: wrong_decision
+    - thinking_time
+    - with prob path_teamwork, factor in the other player (assuming Greedy) when finding the best path
+    Adding:
+    - Take a sub-optimal first action with Boltzmann rational probability
+    """
+
+    def __init__(self, mlp, player_index,
+                 perseverance=0.5, teamwork=0.8, retain_goals=0.8, wrong_decisions=0.02, thinking_prob=0.8,
+                 path_teamwork=0.8, rationality_coefficient=3, prob_pausing=0.5):
+        self.mlp = mlp
+        self.agent_index = player_index
+        self.mdp = self.mlp.mdp
+        self.prev_state = None
+        self.timesteps_stuck = 0  # Count how many times there's a clash with the other player
+        self.dont_drop = False
+        self.prev_motion_goal = None
+        self.prev_best_action = None
+        self.only_take_dispenser_onions = False  # Only used for layout Forced Coordination: when an onion is
+        # deliberately dropped on a shared counter, then from then on only take onions from the dispenser
+        self.only_take_dispenser_dishes = False
+        self.GHM = GreedyHumanModel_pk(self.mlp, player_index=1 - self.agent_index)  # For ToM of other players
+        self.human_model = True
+
+        # "Personality" parameters (within 0 to 1, except rationality_coefficient)
+        self.perseverance = perseverance  # perseverance = 1 means the agent always tries to go where it want to do, even when it's stuck
+        self.teamwork = teamwork  # teamwork = 0 should make this agent similar to GreedyHuman
+        self.retain_goals = retain_goals  # Prob of keeping the previous goal each timestep (rather than re-calculating)
+        self.wrong_decisions = wrong_decisions  # Prob of making a wrong decision about the motion_goal. I.e. they should
+        # get onion but they get dish instead. Note: this should only work well if retain_goals is large
+        self.thinking_prob = thinking_prob  # After achieving a goal (e.g. fetching onion) the agent waits to "think".
+        # thinking_prob is the probability (p) of moving on during this "thinking time". Expected value of a geometrically
+        # distributed random variable is 1/p, so set p to e.g. 0.25?
+        self.path_teamwork = path_teamwork  # Prob of considering the other agent's location in the path choice
+        self.rationality_coefficient = rationality_coefficient  # Setting to 0 means random actions; inf means always takes
+        # lowest cost path. In practice inf ~ 100
+        self.prob_pausing = prob_pausing # Probability of pausing on a given timestep, instead of acting. From a
+        # quick initial look at the human data, the humans pause very approx 50% of the time
+
+        self.return_action_probs = False  # Don't return action probs
+
+    def reset_agent(self):
```

Comment:
Should be simply called `reset` to conform to the Agent API as defined at the top of the file